src/engine/GodotFileStream.cs

```diff
@@ -17,11 +17,11 @@ public GodotFileStream(File file)
     public override bool CanRead => true;
     public override bool CanSeek => true;
     public override bool CanWrite => true;
-    public override long Length => file.GetLen();
+    public override long Length => checked((long)file.GetLen());
```

Comment:
Would it be cleaner to change our data types regarding file lengths to use ulongs? I don't think we do much match with the file lengths, so I think it should be a safe change.

Reply:
Yes, problem is, we subclass IO.Stream, which needs long.

Reply:
Ah, I see. I guess then this'll have to do. I guess some of the stuff that doesn't follow a fixed interface could be changed, but it would be pretty weird to use half and half of long and ulong for file lengths.